lib/webpacker/railtie.rb

```diff
@@ -24,7 +24,7 @@ class Webpacker::Railtie < ::Rails::Railtie
   #     - add `config.webpacker.check_yarn_integrity = false`
   initializer "webpacker.yarn_check" do |app|
     if app.config.webpacker[:check_yarn_integrity] || (!app.config.webpacker.key?(:check_yarn_integrity) && Rails.env.development?)
-      ok = system("yarn check --integrity")
+      ok = !File.exist?("yarn.lock") || system("yarn check --integrity")
```

Comment:
Bit confused with amount of conditional we have here. Lets just use `app.config.webpacker[:check_yarn_integrity]` and set it to true for development using installer. 

```rb
environment \
  "# Verifies that versions and hashed value of the package contents in the project’s package.json .\n" +
  "  # config.webpacker[:check_yarn_integrity] = true \n",
  env: 'development'

environment \
  "# Verifies that versions and hashed value of the package contents in the project’s package.json .\n" +
  "  # config.webpacker[:check_yarn_integrity] = false \n",
  env: 'production'
```

```rb
# Set it default in Railtie itself
# after config.webpacker = ActiveSupport::OrderedOptions.new
config.webpacker.check_yarn_integrity = false

if File.exist?('yarn.lock') && app.config.webpacker.check_yarn_integrity
  # ... run initializer 
end
```

Reply:
I'll change this following your suggestion. Thanks!

Reply:
I fixed it and confirmed:

#### After `webpacker:install`

```diff
diff --git a/config/environments/development.rb b/config/environments/development.rb
index 376aa7f..85b9ced 100644
--- a/config/environments/development.rb
+++ b/config/environments/development.rb
@@ -1,4 +1,7 @@
 Rails.application.configure do
+  # Verifies that versions and hashed value of the package contents in the project's package.json
+  config.webpacker.check_yarn_integrity = true
+
   # Settings specified here will take precedence over those in config/application.rb.

   # In the development environment your application's code is reloaded on
diff --git a/config/environments/production.rb b/config/environments/production.rb
index 0a7c662..a05bc32 100644
--- a/config/environments/production.rb
+++ b/config/environments/production.rb
@@ -1,4 +1,7 @@
 Rails.application.configure do
+  # Verifies that versions and hashed value of the package contents in the project's package.json
+  config.webpacker.check_yarn_integrity = false
+
   # Settings specified here will take precedence over those in config/application.rb.

   # Code is not reloaded between requests.
```

And on booting server process, it's ok to check integrity:

```bash
% bundle exec rails s
=> Booting Puma
=> Rails 5.2.0.alpha application starting in development
=> Run `rails server -h` for more startup options
yarn check v1.2.1
success Folder in sync.
✨  Done in 0.10s.
Puma starting in single mode...
* Version 3.10.0 (ruby 2.4.2-p198), codename: Russell's Teapot
* Min threads: 5, max threads: 5
* Environment: development
* Listening on tcp://0.0.0.0:3000
Use Ctrl-C to stop
```

Reply:
@gauravtiwari Would you please code-review this again?